src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Telemetry/ITelemetryReporterExtensions.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Razor.Telemetry;
+
+internal static class ITelemetryReporterExtensions
+{
+    // These extensions effectively make TimeSpan an optional parameter on BeginBlock
+    internal static TelemetryScope BeginBlock(this ITelemetryReporter self, string name, Severity severity) { return self.BeginBlock(name, severity, TimeSpan.Zero); }
+    internal static TelemetryScope BeginBlock(this ITelemetryReporter self, string name, Severity severity, Property property) { return self.BeginBlock(name, severity, TimeSpan.Zero, property); }
+    internal static TelemetryScope BeginBlock(this ITelemetryReporter self, string name, Severity severity, Property property1, Property property2) { return self.BeginBlock(name, severity, TimeSpan.Zero, property1, property2); }
+    internal static TelemetryScope BeginBlock(this ITelemetryReporter self, string name, Severity severity, Property property1, Property property2, Property property3) { return self.BeginBlock(name, severity, TimeSpan.Zero, property1, property2 , property3); }
+    internal static TelemetryScope BeginBlock(this ITelemetryReporter self, string name, Severity severity, params ReadOnlySpan<Property> properties) { return self.BeginBlock(name, severity, TimeSpan.Zero, properties); }
```

Comment:
To consider:

- Give a reasonable name to the receiver (`this`) parameter of an extension method. Parameter names like `self` or `@this` can look pretty odd when used as a named argument if the extension method every needs to be invoked as a static method.
- Avoid extremely long lines.
- Use expression-bodied members to enjoy more extra sharp double arrows in your code! (We use them all over the place in Razor.)
- Don't place method declarations and bodies on the same line unless they are a) expression bodied members and b) super short.
- Use named arguments for `TimeSpan.Zero` for readability.

```suggestion
    public static TelemetryScope BeginBlock(this ITelemetryReporter reporter, string name, Severity severity)
        => reporter.BeginBlock(name, severity, minTimeToReport: TimeSpan.Zero);

    public static TelemetryScope BeginBlock(this ITelemetryReporter reporter, string name, Severity severity, Property property)
        => reporter.BeginBlock(name, severity, minTimeToReport: TimeSpan.Zero, property);

    public static TelemetryScope BeginBlock(this ITelemetryReporter reporter, string name, Severity severity, Property property1, Property property2)
        => reporter.BeginBlock(name, severity, minTimeToReport: TimeSpan.Zero, property1, property2);

    public static TelemetryScope BeginBlock(this ITelemetryReporter reporter, string name, Severity severity, Property property1, Property property2, Property property3)
        => reporter.BeginBlock(name, severity, minTimeToReport: TimeSpan.Zero, property1, property2 , property3);

    public static TelemetryScope BeginBlock(this ITelemetryReporter reporter, string name, Severity severity, params ReadOnlySpan<Property> properties)
        => reporter.BeginBlock(name, severity, minTimeToReport: TimeSpan.Zero, properties);
```